src/components/UploadFile.vue

```diff
@@ -150,30 +152,70 @@ export default createComponent({
                 state.fileUint8Array = new Uint8Array(digest);
             }
 
-            //TODO [2019-10-18]: Handle file splitting: if upload fails with some error which indirectly means: "file too big", chop file in half and try again with create for first half and fileAppendTransaction for second half - recursively do these halves until no error
-
             try {
                 state.isBusy = true;
-                const { FileCreateTransaction, Client } = await (import(
-                    "@hashgraph/sdk"
-                ) as Promise<typeof import("@hashgraph/sdk")>);
+                console.log("uploading");
+
+                const {
+                    FileCreateTransaction,
+                    FileAppendTransaction,
+                    Client
+                } = await (import("@hashgraph/sdk") as Promise<
+                    typeof import("@hashgraph/sdk")
+                >);
+
+                const file: Uint8Array = state.fileUint8Array as Uint8Array;
+                const chunks = [];
+
+                const chunkCount = Math.ceil(file.byteLength / MAX_FILE_LENGTH);
+                console.log("file chunks " + chunkCount);
+
+                for (let i = 0; i < chunkCount; i++) {
+                    console.log("chunk " + i);
+
+                    const start = i * MAX_FILE_LENGTH;
+                    chunks.push(file.slice(start, start + MAX_FILE_LENGTH));
+                }
 
                 const receipt = ref<TransactionReceipt | null>(null);
 
+                const publicKey = (await store.state.wallet.session.wallet.getPublicKey()) as import("@hashgraph/sdk").Ed25519PublicKey;
+
+                // TODO[2019-10-18]: send alert to user and back out of interface
+                if (!publicKey) {
+                    throw new Error("not logged in");
+                }
+
                 receipt.value = await new FileCreateTransaction(
                     client as InstanceType<typeof Client>
                 )
-                    .setContents(state.fileUint8Array as Uint8Array)
+                    .setContents(chunks.pop() as Uint8Array)
                     .setExpirationTime(Date.now() + 7890000000)
-                    .addKey(
-                        (await store.state.wallet.session.wallet.getPublicKey()) as import("@hashgraph/sdk").Ed25519PublicKey
-                    )
+                    .addKey(publicKey)
                     .setTransactionFee(1e12)
                     .build()
                     .executeForReceipt();
 
                 const fileId = receipt.value.fileId;
 
+                // TODO[2019-10-18]: handle this properly lol
+                if (!fileId) {
+                    throw new Error("no file id :(");
+                }
+
+                for (const chunk in chunks) {
+                    console.log("chunk!");
+
+                    await new FileAppendTransaction(client as InstanceType<
```

Comment:
Trying .svg file @ 25.9 kB and it errors out with `Error: "Hedera returned response code: FILE_CONTENT_EMPTY (47)"`.